src/tiledimage.js

```diff
@@ -1003,6 +1009,92 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
         });
     },
 
+    /**
+     * Update headers to include when making AJAX requests.
+     *
+     * Unless `propagate` is set to false (which is likely only useful in rare circumstances),
+     * the updated headers are propagated to all tiles and queued image loader jobs.
+     *
+     * Note that the rules for merging headers still apply, i.e. headers returned by
+     * {@link OpenSeadragon.TileSource#getTileAjaxHeaders} take precedence over
+     * the headers here in the tiled image (`TiledImage.ajaxHeaders`).
+     *
+     * @function
+     * @param {Object} ajaxHeaders Updated AJAX headers, which will be merged over any headers specified in {@link OpenSeadragon.Options}.
+     * @param {Boolean} [propagate=true] Whether to propagate updated headers to existing tiles and queued image loader jobs.
+     */
+    setAjaxHeaders: function(ajaxHeaders, propagate) {
+        if (ajaxHeaders === null) {
+            ajaxHeaders = {};
+        }
+        if (!$.isPlainObject(ajaxHeaders)) {
+            console.error('[TiledImage.setAjaxHeaders] Ignoring invalid headers, must be a plain object');
+            return;
+        }
+
+        this._ownAjaxHeaders = ajaxHeaders;
+        this.updateAjaxHeaders(propagate);
+    },
+
+    /**
+     * Update headers to include when making AJAX requests.
+     *
+     * This function has the same effect as calling {@link OpenSeadragon.TiledImage#setAjaxHeaders},
+     * expect that the headers for this tiled image do not change. This is especially useful
+     * for propagating updated headers from {@link OpenSeadragon.TileSource#getTileAjaxHeaders}
+     * to existing tiles.
+     *
+     * Note that `TiledImage.ajaxHeaders` might still change if {@link OpenSeadragon.Viewer#setAjaxHeaders}
+     * was previously called without propagating the updated headers (not recommended).
+     *
+     * @function
+     * @param {Boolean} [propagate=true] Whether to propagate updated headers to existing tiles and queued image loader jobs.
+     */
+    updateAjaxHeaders: function(propagate) {
```

Comment:
Looks like you're making this part of the public API... Did you think of something where that would make sense? If there's no reason for it to be public, I'd rather keep it private to keep the API surface area down.

Reply:
I'm not liking this either, but the whole ajax headers system has unfortunately grown too complicated, and updating them in a fully backwards-compatible way is cumbersome.

I thought it's necessary for the special use case of updating the headers of existing tiles (because `tileSource.getTileAjaxHeaders` has changed), without wanting to change the static headers of the tiled image. In a nutshell, `tiledImage.updateAjaxHeaders()` has the same effect as `tiledImage.setAjaxHeaders(tiledImage._ownAjaxHeaders)`.

An alternative solution is to not provide all these options and just force users to trigger a (propagating) header update at the viewer, even if the viewer's ajax headers haven't changed, i.e. calling `viewer.setAjaxHeaders(viewer.ajaxHeaders)`.

Reply:
Let me know what you prefer.

Reply:
I think `updateAjaxHeaders` is good for clean internals, but I don't think we need to make it public in the API. Like you say, if someone really needs it, they can use `viewer.setAjaxHeaders(viewer.ajaxHeaders)`. 

To make this "private", prefix it with an underscore and add `@private` to the doc comment. We're not enforcing privacy, but those things are enough of a signal to developers that they shouldn't necessarily rely on that function being part of the official API.

Reply:
I made `updateAjaxHeaders` private in 4060dee4f97d1b06df3663e16cbf66eac097b44c.